datafusion/functions/src/crypto/basic.rs

```diff
@@ -157,14 +157,18 @@ pub fn md5(args: &[ColumnarValue]) -> Result<ColumnarValue> {
     })
 }
 
-/// this function exists so that we do not need to pull in the crate hex. it is only used by md5
-/// function below
+/// Hex encoding lookup table for fast byte-to-hex conversion
+const HEX_CHARS_LOWER: &[u8; 16] = b"0123456789abcdef";
+
+/// Fast hex encoding using a lookup table instead of format strings.
+/// This is significantly faster than using `write!("{:02x}")` for each byte.
```

Comment:
It's a bit weird to put the comment here. Maybe put on top of `HEX_CHARS_LOWER` lookup code.